weave/trace_server_bindings/remote_http_trace_server.py

```diff
@@ -123,6 +125,68 @@ def from_env(cls, should_batch: bool = False) -> "RemoteHTTPTraceServer":
     def set_auth(self, auth: t.Tuple[str, str]) -> None:
         self._auth = auth
 
+    def _strip_large_values_from_batch(
+        self, batch: t.List, encoded_bytes: int
+    ) -> t.List:
+        """
+        Iterate through the batch and replace large values with placeholders.
+
+        If keys on inputs or output on start/end reqs are larger than 1MiB,
+        replace them with placeholder values.
+        """
+
+        def _num_bytes(data: t.Any) -> int:
+            """
+            Calculate the number of bytes in a string.
+
+            This can be computationally expensive, only call when necessary.
+            """
+            return len(str(data).encode("utf-8"))
```

Comment:
what about data that does not have a string representation? seems like this could error